manifests/repo.pp

```diff
@@ -0,0 +1,40 @@
+# Configure the Pulpcore repo
+#
+# @param version
+#   The Pulpcore version to use
+class pulpcore::repo (
+  Pattern['^\d+\.\d+$'] $version = '3.7',
+) {
+  $base = "https://yum.theforeman.org/pulpcore/${version}/el${facts['os']['release']['major']}"
+  $baseurl = "${base}/\$basearch"
+
+  $pulpcore_repo_conf = @("REPO")
+    [pulpcore]
+    name=Pulpcore ${version}
+    baseurl=${baseurl}
+    enabled=1
+    gpgcheck=1
+    gpgkey=${base}/GPG-RPM-KEY-pulpcore
+    module_hotfixes=1
+
+    [pulpcore-source]
+    name=Pulpcore ${version} Soure
+    baseurl=${base}/source
+    enabled=0
+    gpgcheck=1
+    gpgkey=${base}/GPG-RPM-KEY-pulpcore
+    module_hotfixes=1
+    | REPO
```

Comment:
Can you share why you chose to inline this here instead of using a file or template?  
Is this puppet DSL version of HEREDOC?

Reply:
I actually copied it from the setup_acceptance_node file without thinking too much. It is indeed the Puppet DSL version of heredoc but a separate template probably makes more sense.